cmd/virt-launcher/virt-launcher.go

```diff
@@ -432,6 +455,9 @@ func main() {
 	// managing virtual machines.
 	markReady()
 
+	// Wait for the libvirt domain to be created.
+	waitForLibvirtDomain(*domainCreationTimeout)
+
 	domain := waitForDomainUUID(*qemuTimeout, events, signalStopChan, domainManager)
```

Comment:
I think these 2 functions are waiting for the same thing.
`waitForDomainUUID` is also waiting for the domain to be defined, not started.
See: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/notify-client/client.go#L253-L256

However, it seems that there may be an issue that under load events are not being fired and that's the reason for timeouts in BZ#1903679

Reply:
Wow you're right, based on the name `qemu-timeout` I assumed it was waiting for qemu...
Thank you, closing this.